test/util/removeUnnecessaryParens.test.js

```diff
@@ -22,7 +22,7 @@ describe('removeUnnecessaryParens', function () {
     ['(((-5)))', '-5'],
     ['((4+5)) + ((2^3))', '(4 + 5) + 2^3'],
     ['(2x^6 + -50 x^2) - (x^4)', '2x^6 - 50x^2 - x^4'],
-    ['(x+4) - (12 + x)', 'x + 4 - (12 + x)'],
+    ['(x+4) - (12 + x)', '(x + 4) - (12 + x)'],
```

Comment:
Simplify does use print, but I'm confused why this would..

OH maybe try cloning the node first and then printing it? 

If that doesn't work, lemme know which tests fail and I can help you look into it